pkg/queue/concurrency_state.go

```diff
@@ -155,6 +154,10 @@ func (c *ConcurrencyEndpoint) Request(action string) error {
 	return nil
 }
 
+func (c *ConcurrencyEndpoint) Terminating(logger *zap.SugaredLogger) {
```

Comment:
thinking out loud (not for this PR): wonder how useful having explicit "pause/resume/terminating" methods is vs. "Notify(ConcurrencyStatePaused)" etc

Reply:
added to todo list